src/main/java/io/supertokens/passwordless/PasswordlessDeviceId.java

```diff
@@ -25,7 +27,14 @@ public PasswordlessDeviceIdHash getHash() throws NoSuchAlgorithmException {
         return new PasswordlessDeviceIdHash(Utils.hashSHA256Bytes(bytes));
     }
 
-    public PasswordlessLinkCode getLinkCode(String userInputCode) throws NoSuchAlgorithmException, InvalidKeyException {
-        return new PasswordlessLinkCode(Utils.hmacSHA256(bytes, userInputCode));
+    public PasswordlessLinkCode getLinkCode(PasswordlessLinkCodeSalt linkCodeSalt, String userInputCode)
+            throws NoSuchAlgorithmException, InvalidKeyException, IOException {
+        // We mix in the salt into the deviceId by concatenating them.
```

Comment:
you should add a comment explain the need for a salt, given that we already have userInputCode, which in theory, can act like a salt too..